sycl/source/device.cpp

```diff
@@ -247,8 +247,11 @@ bool device::ext_oneapi_can_access_peer(const device &peer,
                           "Unrecognized peer access attribute.");
   }();
   auto Plugin = impl->getPlugin();
-  Plugin->call(urUsmP2PPeerAccessGetInfoExp, Device, Peer, UrAttr, sizeof(int),
-               &value, &returnSize);
+  Plugin->call(urUsmP2PPeerAccessGetInfoExp, Device, Peer, UrAttr, 0, nullptr,
+               &returnSize);
+
+  Plugin->call(urUsmP2PPeerAccessGetInfoExp, Device, Peer, UrAttr, returnSize,
+               &value, nullptr);
```

Comment:
@raiyanla , can underlying UR adapter return more than `sizeof(int)` bytes of data here? If so, we'll write beyond `value`'s storage. If not, what was the issue before?

Reply:
L0 Adapter only returns sizeof(int) max. I think the issue was possibly more with querying for both the returnSize and value at same time. I saw it was returning garbage numbers each time until this PR. All the adapters use the same helper in the implementation for this, and I'm guessing it doesn't set value properly if returnSize is being queried for at the same time. I will need to confirm when I have access to an appropriate machine though.